src/commands/convertConnections/convertLocalConnections.ts

```diff
@@ -0,0 +1,65 @@
+import { AzureWizard, type AzureWizardExecuteStep, type AzureWizardPromptStep, type IActionContext, type ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { type MessageItem } from "vscode";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { createActivityContext } from "../../utils/activityUtils";
+import { ConvertSettingsStep } from "./ConvertSettingsStep";
+import { type IConvertConnectionsContext } from "./IConvertConnectionsContext";
+import { SelectConnectionsStep, type Connection } from "./SelectConnectionsStep";
+
+export async function convertLocalConnections(context: IActionContext, node?: AppSettingTreeItem): Promise<void> {
+    const connections: Connection[] = [];
+    if (node instanceof AppSettingTreeItem) {
+        connections.push({ name: node.id, value: node.value })
+        await node.runWithTemporaryDescription(context, localize('converting', 'Converting...'), async () => {
+            await convertLocalConnectionsInternal(context, connections);
+        });
+    } else {
+        await convertLocalConnectionsInternal(context);
+    }
+}
+
+export async function convertLocalConnectionsInternal(context: IActionContext, connections?: Connection[]): Promise<void> {
+    const wizardContext: IActionContext & Partial<IConvertConnectionsContext> = {
+        ...context,
+        local: true,
+        ...await createActivityContext()
+    };
+
+    wizardContext.connections = connections;
+
+    wizardContext.activityChildren = [];
+
+    const title: string = localize('convertLocalConnections', 'Convert Local Project Connections to Identity-Based Connections');
+
+    const promptSteps: AzureWizardPromptStep<IConvertConnectionsContext>[] = [];
+    const executeSteps: AzureWizardExecuteStep<IConvertConnectionsContext>[] = [];
+
+    const subscriptionPromptStep: AzureWizardPromptStep<ISubscriptionActionContext> | undefined = await ext.azureAccountTreeItem.getSubscriptionPromptStep(context);
+    if (subscriptionPromptStep) {
+        promptSteps.push(subscriptionPromptStep);
+    }
+
+    promptSteps.push(new SelectConnectionsStep(), new UserAssignedIdentityListStep())
+    executeSteps.push(new ConvertSettingsStep(), new RoleAssignmentExecuteStep(() => wizardContext.roles));
+
+    const wizard: AzureWizard<IConvertConnectionsContext> = new AzureWizard(wizardContext, {
+        title,
+        promptSteps,
+        executeSteps
+    });
+
+    await wizard.prompt();
+    await wizard.execute();
+
+    const continueOn: MessageItem = { title: localize('continueOn', 'Continue') };
+    const message: string = localize('setConnectionsProperty', 'Successfully converted local connections in order to use identity based connections you may need to set the connection property within your trigger.');
+    await context.ui.showWarningMessage(message, { learnMoreLink: "https://aka.ms/AAuroke", modal: true }, continueOn);
```

Comment:
I believe that we should show this message _before_ they start converting. We can show it again after in the activity log, but if they're worried about messing up their project, it's a little bit late by this point.